### PR TITLE
Hw29

### PR DESCRIPTION
**Pull request by tarkaiev**

### CODE REVIEW

File: src/main/java/ecommerce/filters/AuthorizationFilter.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String requestedUrl = req.getServletPath();
+        if (protectedUrls.get(requestedUrl) == null) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userId == null) {
+            resp.sendRedirect(req.getContextPath() + "/login");
+            return;
```
Comment (TimoxaV) [inline src/main/java/ecommerce/filters/AuthorizationFilter.java]: You should also have this check in AuthenticationFilter. So you can skip it here.

Comment (tarkaiev) [reply src/main/java/ecommerce/filters/AuthorizationFilter.java]: fixed

File: src/main/java/ecommerce/filters/AuthenticationFilter.java [hunk 2]
```diff
@@ -6,11 +6,9 @@
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
-import javax.servlet.annotation.WebFilter;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-@WebFilter(urlPatterns = "/*")
```
Comment (KseniiaMakarova) [inline src/main/java/ecommerce/filters/AuthenticationFilter.java:13]: let's also refactor AuthenticationFilter as proposed in the checklist

File: src/main/java/ecommerce/filters/AuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String requestedUrl = req.getServletPath();
+        if (protectedUrls.get(requestedUrl) == null) {
```
Comment (KseniiaMakarova) [inline src/main/java/ecommerce/filters/AuthorizationFilter.java]: let's put both conditions in one `if`
also, we can use `containsKey()` here

File: src/main/java/ecommerce/filters/AuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/user/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/order/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/complete", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/product/buy", Set.of(Role.RoleName.USER));
```
Comment (KseniiaMakarova) [inline src/main/java/ecommerce/filters/AuthorizationFilter.java]: admin should also have access to:

ManageProductsController
DeleteUserController

user should also have access to:

ShowUserOrdersController
ShowShoppingCartController
RemoveProductFromCartController

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/ecommerce/filters/AuthorizationFilter.java (hunk 3 above)
Comment (KseniiaMakarova) [reply src/main/java/ecommerce/filters/AuthorizationFilter.java]: UPD: to make it work with one `if` you will have to remove `user` local variable, so this is optional

Comment (tarkaiev) [reply src/main/java/ecommerce/filters/AuthorizationFilter.java]: so I can leave as it is?

Comment (tarkaiev) [reply src/main/java/ecommerce/filters/AuthorizationFilter.java]: refactored a bit, using containsKey()

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**